In [2]:
import pandas as pd 
import numpy as np  
import os

In [11]:
paths = []
for root, dirs, files in os.walk(r'C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados'):
    for state in files:
        paths.append(root + '/' + state)

In [12]:
len(paths)



611

Sitios filtrados por Fer

In [4]:
sitios = pd.read_csv(r'C:\Users\Usuario\Desktop\GitHub\DataSets\DataWarehouse\data_sitios.csv')

In [6]:
sitios['gmap_id'].value_counts()

gmap_id
0x80dd2b4c8555edb7:0xfc33d65c4bdbef42    2
0x87e234e2113f19e7:0xe9a2d4f1b3581004    2
0x87530e5dc89088ab:0x85396be731d35f0b    2
0x80db1ba712fc3cbb:0xed35f45d681765db    2
0x52afb3666c43872d:0x4b627cb1a039d4f6    2
                                        ..
0x88e77edf2367b3c5:0xc5837bb347427df9    1
0x8887ed924cb597af:0xccb0f1447e966f06    1
0x87b3dfee8884cda9:0xe212ea7a11dbf0e1    1
0x89d95055c43404cb:0x92334a0a61c1aaef    1
0x89c6b993431d9a19:0xea378de3c317996e    1
Name: count, Length: 139598, dtype: int64

In [6]:
sitios[sitios['gmap_id'] == '0x52afb3666c43872d:0x4b627cb1a039d4f6']

name  \
494  Farleys family restaurant   

                                               address  \
494  Farleys family restaurant, 4899 Miller Trunk H...   

                                   gmap_id   latitude  longitude  \
494  0x52afb3666c43872d:0x4b627cb1a039d4f6  46.827849  -92.19477   

                                              category  avg_rating  \
494  ['American restaurant', 'Bakery', 'Catering fo...         4.1   

     num_of_reviews  
494              38

In [5]:
# Drop duplicated gmap_id
sitios.drop_duplicates(subset='gmap_id', inplace=True)

sitios.reset_index(drop=True, inplace=True)

sitios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139598 entries, 0 to 139597
Data columns (total 15 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   name              139598 non-null  object 
 1   address           139324 non-null  object 
 2   gmap_id           139598 non-null  object 
 3   description       43081 non-null   object 
 4   latitude          139598 non-null  float64
 5   longitude         139598 non-null  float64
 6   category          139598 non-null  object 
 7   avg_rating        139598 non-null  float64
 8   num_of_reviews    139598 non-null  int64  
 9   price             62364 non-null   object 
 10  hours             120867 non-null  object 
 11  MISC              138439 non-null  object 
 12  state             121686 non-null  object 
 13  relative_results  112959 non-null  object 
 14  url               139598 non-null  object 
dtypes: float64(3), int64(1), object(11)
memory usage: 16.0+ MB


Prueba de concepto: Funcion filtrado

In [34]:
data = pd.read_json(paths[20], lines=True)
data.tail()

user_id            name           time  rating  \
149995  1.045885e+20   Anne Caldwell  1603210583483       4   
149996  1.090443e+20  Barbara Frigon  1619200281641       5   
149997  1.016523e+20  Gloria Handley  1605727707642       5   
149998  1.048104e+20    Alyssa Stark  1608779865223       4   
149999  1.010976e+20   evelyn batres  1602965500019       5   

                                                     text  pics  resp  \
149995  The renewal process was easy and I don't have ...  None  None   
149996  We have had Farmers Insurance for many years a...  None  None   
149997  I really trust Farmers and my agent.  They hav...  None  None   
149998           Super helpful agents who always respond.  None  None   
149999  My Agent is very helpful and is always availab...  None  None   

                                      gmap_id  
149995  0x872b07d6f10f4f99:0xc064771c28929a9d  
149996  0x872b07d6f10f4f99:0xc064771c28929a9d  
149997  0x872b07d6f10f4f99:0xc064771c28929a9d  
149998  0x872b07d6f10f4f99:0xc064771c28929a9d  
149999  0x872b07d6f10f4f99:0xc064771c28929a9d

In [24]:
data.columns

Index(['user_id', 'name', 'time', 'rating', 'text', 'pics', 'resp', 'gmap_id'], dtype='object')

In [25]:
data.shape

(150000, 8)

In [26]:
sitios.shape

(139598, 8)

In [35]:
data.merge(sitios[['gmap_id', 'name', 'avg_rating', 'category']],
           how='inner', on='gmap_id')

user_id                    name_x           time  rating  \
0     1.089615e+20              Emily Garcia  1389417567515       1   
1     1.125856e+20                Tjad Olson  1442729477996       1   
2     1.089615e+20              Emily Garcia  1389417567515       1   
3     1.029550e+20                  Vgasbaby  1432255837805       1   
4     1.012840e+20  Alexandria Schmid (Alex)  1404506737211       2   
...            ...                       ...            ...     ...   
4527  1.091376e+20               Nancy Jiang  1504892411034       3   
4528  1.133612e+20            Alyssa Galello  1517165599529       5   
4529  1.039963e+20                    Jake E  1503599944563       5   
4530  1.045278e+20                 Jim Roker  1519918429763       4   
4531  1.010759e+20           carlos oceguera  1498990695253       5   

                                                   text  pics  resp  \
0     This place was not a good pick for Happy Hour....  None  None   
1     They put my beer in a mug. That's it! Good liv...  None  None   
2     This place was not a good pick for Happy Hour....  None  None   
3     The bartenders here have no respect. She slamm...  None  None   
4     Stopped in on a Sunday evening around 10:20 pm...  None  None   
...                                                 ...   ...   ...   
4527                                               None  None  None   
4528                                               None  None  None   
4529                                               None  None  None   
4530                                               None  None  None   
4531                                               None  None  None   

                                    gmap_id               name_y  avg_rating  \
0     0x872baabc8feba081:0xa268179387bfff11  Whiskey Rose Saloon         2.3   
1     0x872baabc8feba081:0xa268179387bfff11  Whiskey Rose Saloon         2.3   
2     0x872baabc8feba081:0xa268179387bfff11  Whiskey Rose Saloon         2.3   
3     0x872baabc8feba081:0xa268179387bfff11  Whiskey Rose Saloon         2.3   
4     0x872baabc8feba081:0xa268179387bfff11  Whiskey Rose Saloon         2.3   
...                                     ...                  ...         ...   
4527   0x86d665610d314e51:0x1a894e07e756b0c        Mr. Baja Fish         4.1   
4528   0x86d665610d314e51:0x1a894e07e756b0c        Mr. Baja Fish         4.1   
4529   0x86d665610d314e51:0x1a894e07e756b0c        Mr. Baja Fish         4.1   
4530   0x86d665610d314e51:0x1a894e07e756b0c        Mr. Baja Fish         4.1   
4531   0x86d665610d314e51:0x1a894e07e756b0c        Mr. Baja Fish         4.1   

                                  category  
0                   ['Bar & grill', 'Bar']  
1                   ['Bar & grill', 'Bar']  
2                   ['Bar & grill', 'Bar']  
3                   ['Bar & grill', 'Bar']  
4                   ['Bar & grill', 'Bar']  
...                                    ...  
4527  ['Mexican restaurant', 'Restaurant']  
4528  ['Mexican restaurant', 'Restaurant']  
4529  ['Mexican restaurant', 'Restaurant']  
4530  ['Mexican restaurant', 'Restaurant']  
4531  ['Mexican restaurant', 'Restaurant']  

[4532 rows x 11 columns]

In [12]:
alabama = pd.read_csv(
    './DataSets/DataWarehouse/Alabama.csv'
    )

In [14]:
alabama.merge(sitios[['gmap_id', 'name', 'avg_rating', 'category']],
           how='inner', on='gmap_id')

Unnamed: 0                user_id                              name_x  \
0              585  114566479906039707843                       Karen Peacock   
1              586  112536090555782546236                            A. Clark   
2              587  101645593904662038896                         Keith Hobbs   
3              588  115454614235191544643                         Martha Slay   
4              589  107363906870632974411  Amanda Susanne Ledbetter Lassitter   
...            ...                    ...                                 ...   
110158      715987  104083846122359953657                       Vicki Langham   
110159      715988  106831022164549779794                       Tora Brookins   
110160      715989  107517137385130933158                         Ethan Widen   
110161      715990  111271142462597249721                      Gaines Harrell   
110162      715991  116832656691098209136                          Shone Dodd   

                 time  rating  \
0       1629127631455       5   
1       1629601387431       2   
2       1628262212791       5   
3       1630466890240       5   
4       1628983539107       5   
...               ...     ...   
110158  1618956516837       4   
110159  1620558482987       5   
110160  1622175274331       5   
110161  1621022828367       5   
110162  1619277967888       4   

                                                     text  \
0       The Tuesday trivia night and Rock n' Roll Bing...   
1       Newly remodeled. Open to the Public. Outside l...   
2       Last night was my first time back since the re...   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
110158       Great atmosphere,  very relaxing and quaint.   
110159                                                NaN   
110160                                                NaN   
110161                                                NaN   
110162                                                NaN   

                                      gmap_id  \
0       0x889a43f81e27b2a1:0x76ac4d38601d38e9   
1       0x889a43f81e27b2a1:0x76ac4d38601d38e9   
2       0x889a43f81e27b2a1:0x76ac4d38601d38e9   
3       0x889a43f81e27b2a1:0x76ac4d38601d38e9   
4       0x889a43f81e27b2a1:0x76ac4d38601d38e9   
...                                       ...   
110158  0x888e8fe674183e3b:0x5801089d63c7ddca   
110159  0x888e8fe674183e3b:0x5801089d63c7ddca   
110160  0x888e8fe674183e3b:0x5801089d63c7ddca   
110161  0x888e8fe674183e3b:0x5801089d63c7ddca   
110162  0x888e8fe674183e3b:0x5801089d63c7ddca   

                                    name_y  avg_rating        category  
0        Lake Forest Yacht Club Restaurant         4.6  ['Restaurant']  
1        Lake Forest Yacht Club Restaurant         4.6  ['Restaurant']  
2        Lake Forest Yacht Club Restaurant         4.6  ['Restaurant']  
3        Lake Forest Yacht Club Restaurant         4.6  ['Restaurant']  
4        Lake Forest Yacht Club Restaurant         4.6  ['Restaurant']  
...                                    ...         ...             ...  
110158  West Main Coffeehouse and Creamery         4.9        ['Cafe']  
110159  West Main Coffeehouse and Creamery         4.9        ['Cafe']  
110160  West Main Coffeehouse and Creamery         4.9        ['Cafe']  
110161  West Main Coffeehouse and Creamery         4.9        ['Cafe']  
110162  West Main Coffeehouse and Creamery         4.9        ['Cafe']  

[110163 rows x 10 columns]

Prueba loop sobre directorio de reviews entero
---

In [8]:
len(paths)

0

Hay un total de 611 archivos para leer y procesar, eliminando los registros que contengan un gmap_id no coincidente con la categoría elegida

In [13]:
lenths = []
filas_encontradas=[]
for path in paths:
    df = pd.read_json(path, lines=True, encoding='utf-8')
    df = df.merge(sitios, how='inner', on='gmap_id')
    
    lenght = df.shape[0]
    print(f'Archivo: {path}\n lenght: {lenght}')
    if lenght > 0:
        filas_encontradas.append(df)
    #lenths.append(lenght)
df_consumo_final=pd.concat(filas_encontradas, ignore_index=True)


# Eliminar columnas innecesarias
columnas_a_eliminar = ["time", "pics", "resp"]
df_consumo_final.drop(columns=columnas_a_eliminar, inplace=True, errors='ignore')
df_consumo_final["text"]=df_consumo_final["text"].map(lambda txt: txt.strip(), na_action="ignore").fillna("No comment")
df_consumo_final["text"]=df_consumo_final["text"].map(lambda txt: txt.replace("\n", " "), na_action="ignore")


df_consumo_final.to_csv(r"C:\Users\Usuario\Desktop\GitHub\DataSets\DataWarehouse\Google-CF.csv", index=False)


Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/1.json
 lenght: 8123
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/10.json
 lenght: 0
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/11.json
 lenght: 0
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/12.json
 lenght: 0
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/2.json
 lenght: 29862
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/3.json
 lenght: 44408
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/4.json
 lenght: 14319
Archivo: C:\Users\Usuario\Desktop\GitHub\DataSets\DataLake\REVIEW-ESTADOS\reviews-estados\review-Alabama/5.json
 lenght: 

In [10]:
df_consumo_final.head(5)

user_id                              name_x  rating  \
0  1.145665e+20                       Karen Peacock       5   
1  1.125361e+20                            A. Clark       2   
2  1.016456e+20                         Keith Hobbs       5   
3  1.154546e+20                         Martha Slay       5   
4  1.073639e+20  Amanda Susanne Ledbetter Lassitter       5   

                                                text  \
0  The Tuesday trivia night and Rock n' Roll Bing...   
1  Newly remodeled. Open to the Public. Outside l...   
2  Last night was my first time back since the re...   
3                                         No comment   
4                                         No comment   

                                 gmap_id                             name_y  \
0  0x889a43f81e27b2a1:0x76ac4d38601d38e9  Lake Forest Yacht Club Restaurant   
1  0x889a43f81e27b2a1:0x76ac4d38601d38e9  Lake Forest Yacht Club Restaurant   
2  0x889a43f81e27b2a1:0x76ac4d38601d38e9  Lake Forest Yacht Club Restaurant   
3  0x889a43f81e27b2a1:0x76ac4d38601d38e9  Lake Forest Yacht Club Restaurant   
4  0x889a43f81e27b2a1:0x76ac4d38601d38e9  Lake Forest Yacht Club Restaurant   

                                             address   latitude  longitude  \
0  Lake Forest Yacht Club Restaurant, 1 Yacht Clu...  30.637477 -87.916952   
1  Lake Forest Yacht Club Restaurant, 1 Yacht Clu...  30.637477 -87.916952   
2  Lake Forest Yacht Club Restaurant, 1 Yacht Clu...  30.637477 -87.916952   
3  Lake Forest Yacht Club Restaurant, 1 Yacht Clu...  30.637477 -87.916952   
4  Lake Forest Yacht Club Restaurant, 1 Yacht Clu...  30.637477 -87.916952   

         category  avg_rating  num_of_reviews  
0  ['Restaurant']         4.6               8  
1  ['Restaurant']         4.6               8  
2  ['Restaurant']         4.6               8  
3  ['Restaurant']         4.6               8  
4  ['Restaurant']         4.6               8

In [12]:
df_consumo_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7297665 entries, 0 to 7297664
Data columns (total 12 columns):
 #   Column          Dtype  
---  ------          -----  
 0   user_id         float64
 1   name_x          object 
 2   rating          int64  
 3   text            object 
 4   gmap_id         object 
 5   name_y          object 
 6   address         object 
 7   latitude        float64
 8   longitude       float64
 9   category        object 
 10  avg_rating      float64
 11  num_of_reviews  int64  
dtypes: float64(4), int64(2), object(6)
memory usage: 668.1+ MB
